include/pybind11/pytypes.h

```diff
@@ -890,6 +891,23 @@ class iterator : public object {
     object value = {};
 };
 
+class type : public object {
+public:
+    PYBIND11_OBJECT_DEFAULT(type, object, PyType_Check);
```

Comment:
`DEFAULT`? Then you do allow for a default constructor with a `PyObject*` `nullptr`? Is there any use to allowing that, and not just have `PYBIND11_OBJECT`?

Also, this could be `PYBIND11_OBJECT_CVT` (given the right conversion function; since `Py_TYPE` is a macro, we might need a minimal wrapper), if you want to also allow `py::type(obj)` rather than `py::type::of(obj)`?

Reply:
That wóuld allow for non-`explicit` conversions `py::type thisType = obj;`, though. So maybe not what we want, then? :-/ But the extra constructor just like with `py::str` would be nice, though.

Reply:
Then just an `explicit type(handle)` constructor?